src/coldcard.js

```diff
@@ -507,107 +516,177 @@ export class ColdcardSignMultisigTransaction extends ColdcardInteraction {
+   * {
+   *    name: MeMyself,
+   *    quorum: {
+   *      totalSigners: 4,
+   *      requiredSigners: 2,   
+   *    },
+   *    addressType: 'P2WSH',
+   *    network: 'testnet',
+   *    extendedPublicKeys: [
+   *      {
+   *        xfp: 'D0CFA66B',
+   *        xpub: 'tpubD9429UXFGCTKJ9NdiNK4rC5...DdP9'
+   *      },
+   *      {
+   *        xfp: '8E697B74',
+   *        xpub: 'tpubD97nVL37v5tWyMf9ofh5rzn...XgSc',
+   *      },
+   *      {
+   *        xfp: 'BE26B07B',
+   *        xpub: 'tpubD9ArfXowvGHnuECKdGXVKDM...FxPa',
+   *      },
+   *      {
+   *        xfp: '4369050F',
+   *        xpub: 'tpubD8NXmKsmWp3a3DXhbihAYbY...9C8n',
+   *      }
+   *    ]
+   * }
+   */
+export function parseColdcardConfig(configString) {
+    if (typeof configString !== 'string') {
+      throw new TypeError(
+        `ColdCard config must be a string, instead received ${typeof configString}`
+      );
     }
-
-    if (this.jsonConfig.extendedPublicKeys &&
-      this.jsonConfig.extendedPublicKeys.every((xpub) => {
-        // For each xpub, check that xfp exists, the length is 8, type is string, and valid hex
-        if (!xpub.xfp || xpub.xfp === "Unknown") {
-          throw new Error("ExtendedPublicKeys missing at least one xfp.");
+    
+    const options = {};
+    // split the text by line
+    const lines = 
+      configString
+        .split('\n')
+        // formatting
+        .map(line => line.trim())
+        // ignore comments
+        .filter(line => line[0] !== '#' && line.length);
+
+    // for each line separate key and values by colon
+    lines.forEach(line => {
+      // trim white space and set to lowercase
+      const [key, value] = line.split(':').map(str => str.trim());
+      // handle recognized keys (name, format, derivation, and policy)
+      switch (key.toLowerCase()) {
+        case 'name':
+          options.name = value;
+          break;
+        case 'format':
+          options.addressType = value.toUpperCase();
+          break;
+        case 'derivation': {
+          const pathError = validateBIP32Path(value);
+          assert(!pathError.length, pathError);
+          options.derivation = value;
+          break;
         }
-        if (typeof xpub.xfp !== 'string') {
-          throw new Error("XFP not a string");
+        case 'policy': {
+          let [m, n] = value.split(' of ');
+          m = Number(m);
+          // confirm we have valid numbers for m and n
+          assert(!isNaN(m) && !isNaN(n), 'Invalid policy in coldcard config');
+          options.requiredSigners = m;
+          break;
         }
-        if (xpub.xfp.length !== 8) {
-          throw new Error("XFP not length 8");
+        default: {
+          // handle possible xfp: xpub pairs from config
+          try {
+            // will throw if not valid base58 
+            ExtendedPublicKey.fromBase58(value);
+            
+            // check if the key is a valid fingerprint
+            validateRootFingerprint(key);
```

Comment:
i think it does make sense to make sure it's not in there twice rather than lettin em get all the way to the coldcard to find that out.  in caravan we could also edit it to not allow it

Reply:
I agree. I think that's one nice thing about having the individual wallet client config methods kind of separate is that it makes sense to enforce the same things coldcard will enforce (as best we can).

Reply:
Added new check in `validateExtendedPublicKeys` which gets called when instantiating the config.